src/prefect/agent/vertex/agent.py

```diff
@@ -0,0 +1,279 @@
+            layer. Leave blank for no api server (default).
+        - no_cloud_logs (bool, optional): Disable logging to a Prefect backend
+            for this agent and all deployed flow runs. Defaults to `False`.
+        - region_name: (str)
+            Region the job is running in for the endpoint
+
+
+    """
+
+    def __init__(  # type: ignore
+        self,
+        project: str,
+        region_name: str,
+        agent_config_id: str = None,
+        name: str = None,
+        labels: Iterable[str] = None,
+        env_vars: dict = None,
+        max_polls: int = None,
+        agent_address: str = None,
+        no_cloud_logs: bool = None,
+    ) -> None:
+        super().__init__(
+            agent_config_id=agent_config_id,
+            name=name,
+            labels=labels,
+            env_vars=env_vars,
+            max_polls=max_polls,
+            agent_address=agent_address,
+            no_cloud_logs=no_cloud_logs,
+        )
+        self.project = project
+        self.region_name = region_name
+        from google.cloud import aiplatform
+
+        # Initialize client that will be used to create and send requests.
+        # This client only needs to be created once, and can be reused for multiple requests.
+        self.client_options = {
+            "api_endpoint": f"{self.region_name}-aiplatform.googleapis.com"
+        }
+        self.vertex_client = aiplatform.gapic.JobServiceClient(
+            client_options=self.client_options
+        )
+
+    def deploy_flow(self, flow_run: GraphQLResult) -> str:
+        """
+        Deploy a flow run as an Vertex task.
+
+        Args:
+            - flow_run (GraphQLResult): A GraphQLResult flow run object
+
+        Returns:
+            - str: Information about the deployment
+        """
+        run_config = self._get_run_config(flow_run, VertexRun)
+        assert isinstance(run_config, VertexRun)  # mypy
+        taskdef = self.generate_task_definition(flow_run, run_config)
+        parent = f"projects/{self.project}/locations/{self.region_name}"
+
+        self.logger.debug(
+            "Created task definition for flow %s",
+            flow_run.flow.id,
+        )
+
+        resp = self.vertex_client.create_custom_job(parent=parent, custom_job=taskdef)
+
+    def generate_task_definition(
+        self, flow_run: GraphQLResult, run_config: VertexRun
+    ) -> Dict[str, Any]:
+        """Generate an Vertex task definition from a flow run
+
+        Args:
+            - flow_run (GraphQLResult): A flow run object
+            - run_config (VertexRun): The flow's run config
+
+        Returns:
+            - dict: a dictionary representation of an Vertex task definition
+        """
+
+        slug = slugify.slugify(
+            flow_run.flow.name,
+            max_length=255 - len("prefect-"),
+            word_boundary=True,
+            save_order=True,
+        )
+
+        # Set flow image
+        run_config.image = None
```

Comment:
I don't think we want this, I believe this is preventing users from specifying a different image on the config. We should check if the default from the utils will work if we use a non docker storage though

Reply:
I think that was there from when I was dealing with the environment variable limit. Yeah, not sure how it would work without docker storage.